### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by smacker**
Fixes: https://github.com/src-d/code-annotation/issues/20

* OAuth2 with github
* JWT
* Keep token in local storage
* Removed service worker (it catches all locations and I don't see how
to override it without create-react-app eject)

TBD:
- tests
- align with backend work by @dpordomingo

### CODE REVIEW

**Review by dpordomingo — APPROVED** [review]
I focused on reviewing the front part, and since it helps you to keep working, LGTM.
I can rearrange the API part once https://github.com/src-d/code-annotation/pull/16 has been approved

File: server/service/jwt.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+type jwtClaim struct {
+	ID int
+	jwt.StandardClaims
+}
+
+// MakeToken generates token string for a user
+func (j *JWT) MakeToken(user *model.User) (string, error) {
+	claims := &jwtClaim{ID: user.ID}
+	t := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+	fmt.Println("skey", j.signingKey)
```
Comment (carlosms) [inline server/service/jwt.go]: remove

**Comment by carlosms** [comment]
I only have two comments:

- There are secrets that need to be provided, so the documentation should explain what and where to put them. For instance, when the docker image is created to test things locally.
- Right now there isn't a logout functionality, and maybe that was not part of this effort, but it should be added at some point.

**Review by carlosms — APPROVED** [review]

**Comment by smacker** [comment]
@carlosms very good points!

1) I'll update README as soon as we decide to proceed with environment variables not flags. @bzz @dpordomingo please confirm
2) We discussed it offline. Solution: Logout only on FE (because JWT doesn't have logout out of the box). But we need "logout button" on wireframes. @bzz @dpordomingo please approve or disapprove. @ricardobaeta please add logout on wireframes.

**Comment by smacker** [comment]
blocked by #16 and structure review.

**Review by dpordomingo — APPROVED** [review]
I'd not block this PR because of https://github.com/src-d/code-annotation/pull/16
imho this can be merged, and I'll rebase and rework on top of this one. I think it will be better because this functionality is already "completed" from the back to the FE

**Comment by bzz** [comment]
Sounds good, let @smacker finish last few things in here i.e documenting in README how the user should create/provide token \w minimal permissions and set it for a Docker image built locally and then merge this one first.

**Review by bzz — APPROVED** [review]

**Comment by smacker** [comment]
1. redundant `fmt.Println()` is removed
2. added action `logout` (but no link yet)
3. documentation is updated
3. Tests will be written later because the code is about to change a lot after #16 

Merging.

**Pull request merged**